src/components/Profile/ProfileInputField.jsx

```diff
@@ -0,0 +1,98 @@
+// React Imports
+import React, { useState } from 'react';
+// Inrupt Imports
+import { useSession } from '@inrupt/solid-ui-react';
+// Material UI Imports
+import Box from '@mui/material/Box';
+import Button from '@mui/material/Button';
+import CheckIcon from '@mui/icons-material/Check';
+import ClearIcon from '@mui/icons-material/Clear';
+import EditIcon from '@mui/icons-material/Edit';
+import Input from '@mui/material/Input';
+import InputLabel from '@mui/material/InputLabel';
+// Utility Imports
+import { fetchProfileInfo, updateProfileInfo } from '../../model-helpers';
+
+/**
+ * messageFolderProps is an object that stores the props for the MessageFolder
+ * component
+ *
+ * @typedef profileInputFieldProps
+ * @type {object}
+ * @property {string} inputName - Name of input field
+ * @property {string} inputValue - Value of input field used for updating profile
+ * @property {(value: React.SetStateAction<null>) => void} setInputValue - Set
+ * function for inputValue
+ * @property {() => void} fetchProfileData - Handler function for fetching data
+ * from user's profile card
+ * @memberof typedefs
+ */
+
+/**
+ * ProfileInputField Component - Component that creates the editable inputs fields
+ * for the Profile page
+ *
+ * @memberof Inbox
+ * @name ProfileInputField
+ * @param {profileInputFieldProps} Props - Props used for NewMessage
+ * @returns {React.JSX.Element} React component for NewMessage
+ */
+const ProfileInputField = ({ inputName, inputValue, setInputValue, fetchProfileData }) => {
+  const { session } = useSession();
+  const [edit, setEdit] = useState(false);
+
+  const handleUpdateProfile = async (event) => {
+    event.preventDefault();
+    const profileData = await fetchProfileInfo(session);
+    const inputField = inputName === 'Name' ? 'profileName' : inputName.toLowerCase();
+
+    await updateProfileInfo(session, profileData, inputField, inputValue);
+
+    fetchProfileData();
+    setEdit(false);
+  };
+
+  const handleCancelEdit = () => {
+    fetchProfileData();
+
+    setEdit(!edit);
+  };
+
+  const handleEditInput = () => {
+    setEdit(!edit);
+  };
+
+  return (
+    <form onSubmit={handleUpdateProfile}>
+      <Box sx={{ display: 'flex', alignItems: 'center', gap: '10px' }}>
+        <Box sx={{ display: 'flex', alignItems: 'center', gap: '10px' }}>
+          <InputLabel htmlFor={`input-${inputName}`} sx={{ color: 'black' }}>
+            {inputName}:{' '}
+          </InputLabel>
+          <Input
+            id={`input-${inputName}`}
+            value={inputValue === null ? 'No value set' : inputValue}
```

Comment:
Piggybacking on Tim's earlier comment, I'm not positive this is what we want. 
BUT I will gladly admit my understanding can be completely wrong.

---

I don't think the value should ever be `No value set`.  I think that makes a great placeholder though.
I feel this should be `value={inputValue || ''}` and also a `placeholder={inputValue || 'No value set'}`

---

Also the current way it is, you can make the name blank - but there is an error if you try to make the organization blank.
1) gotta deal with error (the code above seems to be fine for that)
2) Name should be required
3) Organization - maybe if in edit mode it should be required (so can't be saved as blank), but still can be cancelled?

Reply:
I think that'll be good for `value` and `placeholder`. I'll have this updated to use:

```
value={inputValue}
placeholder={inputValue || 'No value set'}
```

As for Organization, I could be misunderstanding the issue, but I could save blanks for company name. If the name is blank, it's basically the same as no value set.

https://github.com/codeforpdx/PASS/assets/14917816/768c9719-6270-4f40-9d85-f5f3a287d148